### PR TITLE
MISUV-8336-add-user-session-api-fs

### PR DESCRIPTION
**Pull request by simondrugan16**
added FS functionality

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -179,6 +179,7 @@ feature-switch {
   enable-opt-out = false
   enable-income-sources-new-journey = false
   enable-time-machine = false
+  enable-user-session-API = true
```
Comment (avelx) [inline conf/application.conf:182]: can we use different name for this FS? something more expressive like, enable-session-data-storage or similar

File: app/config/featureswitch/UserSessionApiFS.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config.featureswitch
+
+import config.FrontendAppConfig
+
+trait UserSessionApiFS {
```
Comment (avelx) [inline app/config/featureswitch/UserSessionApiFS.scala:21]: no need to create trait here, why we need this trait btw?

Comment (simondrugan16) [reply app/config/featureswitch/UserSessionApiFS.scala:21]: just how we did it for the other feature switches, i can remove it

File: conf/application.conf (hunk 1 above)
Comment (simondrugan16) [reply conf/application.conf:182]: ya sure, this was just the one originally on the ticket

File: it/test/helpers/agent/ComponentSpecBase.scala [hunk 3]
```diff
@@ -93,6 +93,7 @@ trait ComponentSpecBase extends TestSuite with CustomMatchers
     "microservice.services.contact-frontend.port" -> mockPort,
     "feature-switches.read-from-mongo" -> "false",
     "feature-switch.enable-time-machine" -> "false",
+    "feature-switch.enable-user-session-API" -> "true",
```
Comment (avelx) [inline it/test/helpers/agent/ComponentSpecBase.scala:96]: if you can fix naming here as well: 
_feature-switch.enable-session-data-storage_

Comment (simondrugan16) [reply it/test/helpers/agent/ComponentSpecBase.scala:96]: yep thanks

**Review by avelx — APPROVED** [review]
approved

**Pull request merged**